File: frontend/src/api/audit.ts
import { api } from './client'

export interface AuditLogEntry {
  id: number
  entity_table: string
  entity_id: number
  changed_field: string
  old_value: string | null
  new_value: string | null
  changed_by: string
  changed_at: string | null
}

export function fetchAuditLog(entityTable: string, entityId: number): Promise<AuditLogEntry[]> {
  return api.get<AuditLogEntry[]>(`/audit?entity_table=${entityTable}&entity_id=${entityId}`)
}

File: backend/schemas/audit.py
"""Схеми аудит-логу."""

from pydantic import BaseModel
from typing import Optional


class AuditLogOut(BaseModel):
    id:            int
    entity_table:  str
    entity_id:     int
    changed_field: str
    old_value:     Optional[str]
    new_value:     Optional[str]
    changed_by:    str
    changed_at:    Optional[str]

    model_config = {"from_attributes": True}
